lib/capybara/poltergeist/client/browser.coffee

```diff
@@ -458,22 +460,23 @@ class Poltergeist.Browser
     @current_command.sendResponse(@currentPage.getCustomHeaders())
 
   set_headers: (headers) ->
-    # Workaround for https://code.google.com/p/phantomjs/issues/detail?id=745
-    @currentPage.setUserAgent(headers['User-Agent']) if headers['User-Agent']
-    @currentPage.setCustomHeaders(headers)
+    this.add_headers(headers, false, false)
+
+  add_headers: (headers, local = false, keepExisting = true) ->
+    pages = if local then [@currentPage] else @pages
+    pages.forEach (page) =>
```

Comment:
Since `forEach` is ES5.1 I'm guessing this isn't being tested with PhantomJS 1.9 since I believe it will fail.. Better to use coffeescript comprehensions

Reply:
Hmmm -- appears it is being tested and passing -- could be fine.